x-pack/plugins/apm/public/components/shared/ImpactBar/index.tsx

```diff
@@ -8,13 +8,25 @@ import { EuiProgress } from '@elastic/eui';
 import React from 'react';
 import { StringMap } from '../../../../typings/common';
 
+export interface ImpactItem extends StringMap<any> {
+  impact: number;
+}
+
 // TODO: extend from EUI's EuiProgress prop interface
 interface Props extends StringMap<any> {
   value: number;
+  range: ImpactItem[];
 }
 
-export function ImpactBar({ value, ...rest }: Props) {
+export function ImpactBar({ value, range = [], ...rest }: Props) {
+  const maxImpact = Math.max(0, ...range.map(({ impact }) => impact));
```

Comment:
I think `max={max}` makes a little more sense to me than `range={items}`. When looking at the component from the outside, I was not sure what `range` did. 

There is also the perf consideration that the max will be calculated for every single item.

Reply:
> The same calculation has to happen in one place or the other, though, whether it gets defined outside this component each time it's used or just defined the one time here, which is why I opted to move it in

I don't think that's right. Right now `maxImpact` is defined inside `ImpactBar`, which will be rendered for every item in the list. So if there are 100 items, it will be calculated 100 times. If `maxImpact` is defined outside `ImpactBar` it will only be calculated once, regardless how many items there are.

Additionally I think we need a `minImpact` also. We used to calculate it like this (slightly convoluted)
https://github.com/sqren/kibana/blob/6044cfb463473848af3225d97f95754be5538b5a/x-pack/plugins/apm/public/store/reactReduxRequest/transactionList.js#L17-L21

The `minImpact` exaggerates small differences for high values. 
Eg. If all values are between 90 and 100
Sample value: 91
Previous impact: 10%
New impact: 91%

Sample value: 95
Previous impact: 50%
New impact: 95%

Perhaps it would be simpler to calculate this upfront in node?
https://github.com/elastic/kibana/blob/master/x-pack/plugins/apm/server/lib/transactions/get_top_transactions.js#L87

Reply:
Something like this would work:
```js
function getRelativeImpact(value, min, max) {
  return (value - min) / (max - min)
}
```

Reply:
I see what you're saying now about having to calculate the same value for every item, good catch. The need for a minimum makes sense too. I'll work on something.